pkg/pillar/types/volumetypes.go

```diff
@@ -33,6 +33,11 @@ type VolumeConfig struct {
 
 // Key is volume UUID which will be unique
 func (config VolumeConfig) Key() string {
+	// PVC names should not include # so lets choose -pvc-
+	if base.IsHVTypeKube() {
+		return fmt.Sprintf("%s-pvc-%d", config.VolumeID.String(),
```

Comment:
I'm not sure, but I guess `#` before the generation can be used somewhere in the code for parsing to split the name... Is it possible to have the name with `#%d` at the end?

Reply:
And in general, do we really need the Key of another form for the kubevirt case?.. I may be wrong, but it may be a case in which some other component has an implicit expectation from the Volume[Ref]Config/Status name...

Reply:
Kubernetes by standard does not support names with "#"

Reply:
Did you change the code which parses the name (to extact the generationcounter)?

Reply:
FWIW that code is in
pkg/pillar/cmd/volumemgr/initialvolumestatus.go:		volumeIDAndGeneration = strings.ReplaceAll(volumeIDAndGeneration, "#", ".")

It might be simpler to pick some other unique character (which is not in a UUID nor in a number) over -pvc- so that you can do two reaplaceall e.g., from # and from @ to period in the parser.

Last but not least, you don't need to change the Key() functions at all; those are pubsub keys.
You only need to change the function(s) which form the filename and the above parser of the filename.

And please make sure that the population from the PVC also determines the generationCounter so the VolumeStatus is correct.

Reply:
This inconsistency still seems like the source of a future bug as well as confusion in the naming of the volumes.

Can we use % in a PVC name? If so that would make more sense than -pvc-.
And with appropriate comments and also code in pkg/pillar/cmd/volumemgr/initialvolumestatus.go to treat % as an alternative to #, then I think there is less risk of confusion and bugs.

Reply:
Unfortunately % is also not supported. I tried to create a PVC with name testing-%-0

Resource: "/v1, Resource=persistentvolumeclaims", GroupVersionKind: "/v1, Kind=PersistentVolumeClaim"
Name: "testing-%-0", Namespace: "default"
from server for: "/persist/pvc.yaml": invalid resource name "testing-%-0": [may not contain '%']

Reply:
Actually thinking about it, why does the name even need to have "#" or "%"
Can't we just have UUID-<Generationcounter> ?

Reply:
We could have but changing it now means makes upgrades and fallback hard, because existing devices with existing volumes have "#". We can make cmd/upgradeconverter rename those files in /persist/{vault,clear}/volumes, but what if the upgrade 10 minute test fails and it falls back to the old release? We can't loose the existing volumes in that case.

Reply:
Latest code addresses this. Thanks.